### PR TITLE
Update breaking change docs

### PR DESCRIPTION
**Pull request by BillWagner**
Offline, we had a discussion that the title and H1 were verbose and could be misunderstood.

I've updated the most recent 4 documents to match that discussion.

Notes for reviewers:

All edits in each file should be the same (modulo typos and version numbers). Let's agree on the language on one of them, then make sure they all match.

### CODE REVIEW

**Comment by BillWagner** [comment]
PTAL @jcouv @333fred

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# This document lists known breaking changes in Roslyn after .NET 9 all the way to .NET 10.
+# Roslyn breaking changes from .NET 9.0.100 through .NET 10.0.100
```
Comment (jcouv) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md:1]: I think we should say "after .NET 9.0.100" (9.0.100 is included in previous document, but excluded from this document)

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+
+This document lists known breaking changes in Roslyn after .NET 9 general release (.NET SDK version 9.0.100) through .NET 10 general release (.NET SDK version 10.0.100).
```
Comment (jcouv) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md:3]: As a side note, Bill, I don't remember whether we'd decided on chronological or reverse chronological order for entries in each document. We've been appending at the end (ie. chronological). Does that match your expectation?

Comment (BillWagner) [reply docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md:3]: That works. We've had no concerns on the order.

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
```
Comment (333fred) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md:1]: Is this title going to be updated too?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Review by 333fred — APPROVED** [review]

**Pull request merged**